risc0/circuit/rv32im/src/lib.rs

```diff
@@ -54,3 +57,35 @@ impl TapsProvider for CircuitImpl {
 }
 
 impl CircuitCoreDef<BabyBear> for CircuitImpl {}
+
+/// Fetch a control ID with the given hash, by name, and cycle limit as a power of two (po2) from
+/// the precomputed table. If the hash function is not precomputed, or the po2 is out of range,
+/// this function will return `None`.
+///
+/// Supported values for hash_name are "sha-256", "poseidon2", and "blake2b".
+#[inline]
+pub fn control_id(hash_name: impl AsRef<str>, po2: usize) -> Option<Digest> {
+    if !(risc0_zkp::MIN_CYCLES_PO2..=risc0_zkp::MAX_CYCLES_PO2).contains(&po2) {
+        return None;
+    }
+    let idx = po2 - risc0_zkp::MIN_CYCLES_PO2;
+    match hash_name.as_ref() {
+        "sha-256" => Some(SHA256_CONTROL_IDS[idx]),
+        "poseidon2" => Some(POSEIDON2_CONTROL_IDS[idx]),
+        "blake2b" => Some(BLAKE2B_CONTROL_IDS[idx]),
+        _ => None,
+    }
+}
+
+/// Fetch all precomputed control IDs using the given hash, by name, and up to the cycle limit as a
+/// power of two (po2). If po2 is larger than the max supported, only supported po2s will be
+/// returned.
+pub fn control_ids(
+    hash_name: impl AsRef<str> + 'static,
+    po2_max: usize,
+) -> impl Iterator<Item = Digest> {
```

Comment:
This could instead return `Option<&'static [Digest]>` by doing:
```rs
let all_control_ids = ...; // e.g. SHA256_CONTROL_IDS
let po2_idx = po2_max.checked_sub(risc0_zkp::MIN_CYCLES_PO2)?;
all_control_ids.get(..po2_idx)
```

Reply:
This method is not part of any performance critical path, so I prefer the more flexible and general signature than something optimized for the specific implementation like this.